lib/components/InputBase/withEnhancedInput.tsx

```diff
@@ -60,7 +60,9 @@ export function withEnhancedInput<IncomingProps = {}>(
 		primitiveType: string;
 	},
 	withPrefixIcon = true,
```

Comment:
This is getting a bit out of hand - especially in the consumer side. Do you think we can turn this into an `IconConfig` object.